pyadjoint/tape.py

```diff
@@ -28,9 +27,11 @@ def continue_annotation():
     return _annotation_enabled
 
 
-class set_working_tape(object):
-    """A context manager whithin which a new tape is set as the working tape.
-       This context manager can also be used in an imperative manner.
+class set_working_tape(ContextDecorator):
+    """Set a new tape as the working tape. This can be used either:
+       1) as a free function to replace the working tape,
+       2) as a context manager within which a new tape is set as the working tape
+       3) as a function decorator so that the new tape is set only inside the function.
```

Comment:
```suggestion
    """Set a new tape as the working tape.
    
    This class can be used in three ways:
       1) as a free function to replace the working tape,
       2) as a context manager within which a new tape is set as the working tape,
       3) as a function decorator so that the new tape is set only inside the function.
```

Docstring best practice is to have a short description followed by a blank line.